compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/ClassReference.kt

```diff
@@ -93,38 +94,41 @@ public fun ClassReference.allSuperTypeClassReferences(
   includeSelf: Boolean = false
 ): Sequence<ClassReference> {
 
-  fun KtClassOrObject.superTypeEntryNames() = superTypeListEntries
-    .map { it.requireFqName(module) }
-
-  fun List<FqName>.typeRefs(): Sequence<ClassReference> {
-    return asSequence().map { module.requireClassReference(it) }
+  val initial = if (includeSelf) {
+    sequenceOf(this)
+  } else {
+    directSuperClassReferences(module)
   }
 
-  val initial = sequenceOf(this).drop(if (includeSelf) 0 else 1)
-
   return generateSequence(initial) { superTypes ->
     superTypes
-      .mapNotNull { classRef ->
-        when (classRef) {
-          is Psi -> {
-            classRef.clazz.superTypeEntryNames()
-              .takeIf { it.isNotEmpty() }
-              ?.typeRefs()
-          }
-          is Descriptor -> {
-            classRef.clazz.directSuperClassAndInterfaces()
-              .takeIf { it.isNotEmpty() }
-              ?.asSequence()
-              ?.map { module.requireClassReference(it.fqNameSafe) }
-          }
-        }
-      }
+      .mapNotNull { classRef -> classRef.directSuperClassReferences(module) }
       .flatten()
       .takeIf { it.firstOrNull() != null }
   }.flatten()
     .distinctBy { it.fqName }
```

Comment:
What I meant earlier was the suggestion below and always use the `sequenceOf(this)` as initial value.
```suggestion
  }
    .drop(if (includeSelf) 0 else 1)
    .flatten()
    .distinctBy { it.fqName }
```